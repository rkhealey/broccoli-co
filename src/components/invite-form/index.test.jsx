import _ from 'lodash';
import React from 'react';
import { shallow } from 'enzyme';

import { InviteFormForTest as InviteForm } from './';
import FormSuccess from '../form-success';

describe('components/invite-form', () => {
  const simpleProps = {
    actions: {
      requestInvite: _.noop,
    },
    error: undefined,
    handleSubmit: jest.fn(),
    onClose: jest.fn(),
    submitting: false,
    submitted: false,
    submitFailed: false,
    submitSucceeded: false,
  };

  it('renders with simple props', () => {
    const wrapper = shallow(<InviteForm {...simpleProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with error', () => {
    const wrapper = shallow(<InviteForm {...simpleProps} error="Oops" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with submitted and submitSucceeded', () => {
    const wrapper = shallow(<InviteForm {...simpleProps} submitted submitSucceeded />);
    expect(wrapper.find(FormSuccess)).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('fires the onClose', () => {
    const wrapper = shallow(<InviteForm {...simpleProps} submitted submitSucceeded />);
    wrapper.find(FormSuccess).simulate('click');
    expect(simpleProps.onClose).toHaveBeenCalledTimes(1);
  });

  it('renders with submitFailed', () => {
    const wrapper = shallow(
      <InviteForm {...simpleProps} submitFailed error="Oops" />);
    expect(wrapper.find(FormSuccess)).toHaveLength(0);
    expect(wrapper.find('.errorMsg').text()).toEqual('Oops');
    expect(wrapper).toMatchSnapshot();
  });

  describe('submit form', () => {
    const actions = {
      requestInvite: jest.fn(),
    };
    const formValues = {
      name: 'John Smith',
      email: 'john@test.com',
    };

    const handleSubmit = cb => () => cb(formValues);

    it('submits the form', () => {
      const wrapper = shallow(<InviteForm {...simpleProps} actions={actions} handleSubmit={handleSubmit} />);
      const form = wrapper.find('form').at(0);
      expect(actions.requestInvite).toHaveBeenCalledTimes(0);
      form.simulate('submit');
      expect(actions.requestInvite).toHaveBeenCalledTimes(1);
      expect(actions.requestInvite).toHaveBeenCalledWith({
        name: 'John Smith',
        email: 'john@test.com',
      });
    });
  });
});
