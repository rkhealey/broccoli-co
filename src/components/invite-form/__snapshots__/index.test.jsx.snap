// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/invite-form renders with error 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div>
        <FormHeading>
            Request an invite
        </FormHeading>
        <form
            className="form"
            onSubmit={undefined}
        >
            <div>
                <Field
                    component={[Function]}
                    label="Full Name"
                    name="name"
                    placeholder="Full Name"
                    type="text"
                />
                <Field
                    component={[Function]}
                    label="Email"
                    name="email"
                    placeholder="Email"
                    type="email"
                />
                <Field
                    component={[Function]}
                    label="Confirm email"
                    name="confirmEmail"
                    placeholder="Confirm email"
                    type="email"
                />
            </div>
            <Button
                className="submitBtn"
                onClick={undefined}
                theme="default"
                type="submit"
            >
                Send
            </Button>
        </form>
    </div>
</div>,
  "nodes": Array [
    <div>
      <div>
            <FormHeading>
                  Request an invite
            </FormHeading>
            <form
                  className="form"
                  onSubmit={undefined}
            >
                  <div>
                        <Field
                              component={[Function]}
                              label="Full Name"
                              name="name"
                              placeholder="Full Name"
                              type="text"
                        />
                        <Field
                              component={[Function]}
                              label="Email"
                              name="email"
                              placeholder="Email"
                              type="email"
                        />
                        <Field
                              component={[Function]}
                              label="Confirm email"
                              name="confirmEmail"
                              placeholder="Confirm email"
                              type="email"
                        />
                  </div>
                  <Button
                        className="submitBtn"
                        onClick={undefined}
                        theme="default"
                        type="submit"
                  >
                        Send
                  </Button>
            </form>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <InviteForm
        actions={
                Object {
                        "requestInvite": [Function],
                      }
        }
        error="Oops"
        handleSubmit={[Function]}
        onClose={[Function]}
        submitFailed={false}
        submitSucceeded={false}
        submitted={false}
        submitting={false}
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": InviteForm {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "onSubmit": [Function],
        "props": Object {
          "actions": Object {
            "requestInvite": [Function],
          },
          "error": "Oops",
          "handleSubmit": [Function],
          "onClose": [Function],
          "submitFailed": false,
          "submitSucceeded": false,
          "submitted": false,
          "submitting": false,
        },
        "refs": Object {},
        "renderField": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div>
                    <FormHeading>
                              Request an invite
                    </FormHeading>
                    <form
                              className="form"
                              onSubmit={undefined}
                    >
                              <div>
                                        <Field
                                                  component={[Function]}
                                                  label="Full Name"
                                                  name="name"
                                                  placeholder="Full Name"
                                                  type="text"
                                        />
                                        <Field
                                                  component={[Function]}
                                                  label="Email"
                                                  name="email"
                                                  placeholder="Email"
                                                  type="email"
                                        />
                                        <Field
                                                  component={[Function]}
                                                  label="Confirm email"
                                                  name="confirmEmail"
                                                  placeholder="Confirm email"
                                                  type="email"
                                        />
                              </div>
                              <Button
                                        className="submitBtn"
                                        onClick={undefined}
                                        theme="default"
                                        type="submit"
                              >
                                        Send
                              </Button>
                    </form>
          </div>
</div>,
        "_debugID": 4,
        "_renderedOutput": <div>
          <div>
                    <FormHeading>
                              Request an invite
                    </FormHeading>
                    <form
                              className="form"
                              onSubmit={undefined}
                    >
                              <div>
                                        <Field
                                                  component={[Function]}
                                                  label="Full Name"
                                                  name="name"
                                                  placeholder="Full Name"
                                                  type="text"
                                        />
                                        <Field
                                                  component={[Function]}
                                                  label="Email"
                                                  name="email"
                                                  placeholder="Email"
                                                  type="email"
                                        />
                                        <Field
                                                  component={[Function]}
                                                  label="Confirm email"
                                                  name="confirmEmail"
                                                  placeholder="Confirm email"
                                                  type="email"
                                        />
                              </div>
                              <Button
                                        className="submitBtn"
                                        onClick={undefined}
                                        theme="default"
                                        type="submit"
                              >
                                        Send
                              </Button>
                    </form>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <InviteForm
    actions={
        Object {
            "requestInvite": [Function],
          }
    }
    error="Oops"
    handleSubmit={[Function]}
    onClose={[Function]}
    submitFailed={false}
    submitSucceeded={false}
    submitted={false}
    submitting={false}
/>,
}
`;

exports[`components/invite-form renders with simple props 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div>
        <FormHeading>
            Request an invite
        </FormHeading>
        <form
            className="form"
            onSubmit={undefined}
        >
            <div>
                <Field
                    component={[Function]}
                    label="Full Name"
                    name="name"
                    placeholder="Full Name"
                    type="text"
                />
                <Field
                    component={[Function]}
                    label="Email"
                    name="email"
                    placeholder="Email"
                    type="email"
                />
                <Field
                    component={[Function]}
                    label="Confirm email"
                    name="confirmEmail"
                    placeholder="Confirm email"
                    type="email"
                />
            </div>
            <Button
                className="submitBtn"
                onClick={undefined}
                theme="default"
                type="submit"
            >
                Send
            </Button>
        </form>
    </div>
</div>,
  "nodes": Array [
    <div>
      <div>
            <FormHeading>
                  Request an invite
            </FormHeading>
            <form
                  className="form"
                  onSubmit={undefined}
            >
                  <div>
                        <Field
                              component={[Function]}
                              label="Full Name"
                              name="name"
                              placeholder="Full Name"
                              type="text"
                        />
                        <Field
                              component={[Function]}
                              label="Email"
                              name="email"
                              placeholder="Email"
                              type="email"
                        />
                        <Field
                              component={[Function]}
                              label="Confirm email"
                              name="confirmEmail"
                              placeholder="Confirm email"
                              type="email"
                        />
                  </div>
                  <Button
                        className="submitBtn"
                        onClick={undefined}
                        theme="default"
                        type="submit"
                  >
                        Send
                  </Button>
            </form>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <InviteForm
        actions={
                Object {
                        "requestInvite": [Function],
                      }
        }
        error={undefined}
        handleSubmit={[Function]}
        onClose={[Function]}
        submitFailed={false}
        submitSucceeded={false}
        submitted={false}
        submitting={false}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": InviteForm {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "onSubmit": [Function],
        "props": Object {
          "actions": Object {
            "requestInvite": [Function],
          },
          "error": undefined,
          "handleSubmit": [Function],
          "onClose": [Function],
          "submitFailed": false,
          "submitSucceeded": false,
          "submitted": false,
          "submitting": false,
        },
        "refs": Object {},
        "renderField": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div>
                    <FormHeading>
                              Request an invite
                    </FormHeading>
                    <form
                              className="form"
                              onSubmit={undefined}
                    >
                              <div>
                                        <Field
                                                  component={[Function]}
                                                  label="Full Name"
                                                  name="name"
                                                  placeholder="Full Name"
                                                  type="text"
                                        />
                                        <Field
                                                  component={[Function]}
                                                  label="Email"
                                                  name="email"
                                                  placeholder="Email"
                                                  type="email"
                                        />
                                        <Field
                                                  component={[Function]}
                                                  label="Confirm email"
                                                  name="confirmEmail"
                                                  placeholder="Confirm email"
                                                  type="email"
                                        />
                              </div>
                              <Button
                                        className="submitBtn"
                                        onClick={undefined}
                                        theme="default"
                                        type="submit"
                              >
                                        Send
                              </Button>
                    </form>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <div>
                    <FormHeading>
                              Request an invite
                    </FormHeading>
                    <form
                              className="form"
                              onSubmit={undefined}
                    >
                              <div>
                                        <Field
                                                  component={[Function]}
                                                  label="Full Name"
                                                  name="name"
                                                  placeholder="Full Name"
                                                  type="text"
                                        />
                                        <Field
                                                  component={[Function]}
                                                  label="Email"
                                                  name="email"
                                                  placeholder="Email"
                                                  type="email"
                                        />
                                        <Field
                                                  component={[Function]}
                                                  label="Confirm email"
                                                  name="confirmEmail"
                                                  placeholder="Confirm email"
                                                  type="email"
                                        />
                              </div>
                              <Button
                                        className="submitBtn"
                                        onClick={undefined}
                                        theme="default"
                                        type="submit"
                              >
                                        Send
                              </Button>
                    </form>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <InviteForm
    actions={
        Object {
            "requestInvite": [Function],
          }
    }
    error={undefined}
    handleSubmit={[Function]}
    onClose={[Function]}
    submitFailed={false}
    submitSucceeded={false}
    submitted={false}
    submitting={false}
/>,
}
`;

exports[`components/invite-form renders with submitFailed 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div>
        <FormHeading>
            Request an invite
        </FormHeading>
        <form
            className="form"
            onSubmit={undefined}
        >
            <div>
                <Field
                    component={[Function]}
                    label="Full Name"
                    name="name"
                    placeholder="Full Name"
                    type="text"
                />
                <Field
                    component={[Function]}
                    label="Email"
                    name="email"
                    placeholder="Email"
                    type="email"
                />
                <Field
                    component={[Function]}
                    label="Confirm email"
                    name="confirmEmail"
                    placeholder="Confirm email"
                    type="email"
                />
            </div>
            <Button
                className="submitBtn"
                onClick={undefined}
                theme="default"
                type="submit"
            >
                Send
            </Button>
            <p
                className="errorMsg"
            >
                Oops
            </p>
        </form>
    </div>
</div>,
  "nodes": Array [
    <div>
      <div>
            <FormHeading>
                  Request an invite
            </FormHeading>
            <form
                  className="form"
                  onSubmit={undefined}
            >
                  <div>
                        <Field
                              component={[Function]}
                              label="Full Name"
                              name="name"
                              placeholder="Full Name"
                              type="text"
                        />
                        <Field
                              component={[Function]}
                              label="Email"
                              name="email"
                              placeholder="Email"
                              type="email"
                        />
                        <Field
                              component={[Function]}
                              label="Confirm email"
                              name="confirmEmail"
                              placeholder="Confirm email"
                              type="email"
                        />
                  </div>
                  <Button
                        className="submitBtn"
                        onClick={undefined}
                        theme="default"
                        type="submit"
                  >
                        Send
                  </Button>
                  <p
                        className="errorMsg"
                  >
                        Oops
                  </p>
            </form>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <InviteForm
        actions={
                Object {
                        "requestInvite": [Function],
                      }
        }
        error="Oops"
        handleSubmit={[Function]}
        onClose={[Function]}
        submitFailed={true}
        submitSucceeded={false}
        submitted={false}
        submitting={false}
/>,
      "_debugID": 9,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": InviteForm {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "onSubmit": [Function],
        "props": Object {
          "actions": Object {
            "requestInvite": [Function],
          },
          "error": "Oops",
          "handleSubmit": [Function],
          "onClose": [Function],
          "submitFailed": true,
          "submitSucceeded": false,
          "submitted": false,
          "submitting": false,
        },
        "refs": Object {},
        "renderField": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 5,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div>
                    <FormHeading>
                              Request an invite
                    </FormHeading>
                    <form
                              className="form"
                              onSubmit={undefined}
                    >
                              <div>
                                        <Field
                                                  component={[Function]}
                                                  label="Full Name"
                                                  name="name"
                                                  placeholder="Full Name"
                                                  type="text"
                                        />
                                        <Field
                                                  component={[Function]}
                                                  label="Email"
                                                  name="email"
                                                  placeholder="Email"
                                                  type="email"
                                        />
                                        <Field
                                                  component={[Function]}
                                                  label="Confirm email"
                                                  name="confirmEmail"
                                                  placeholder="Confirm email"
                                                  type="email"
                                        />
                              </div>
                              <Button
                                        className="submitBtn"
                                        onClick={undefined}
                                        theme="default"
                                        type="submit"
                              >
                                        Send
                              </Button>
                              <p
                                        className="errorMsg"
                              >
                                        Oops
                              </p>
                    </form>
          </div>
</div>,
        "_debugID": 10,
        "_renderedOutput": <div>
          <div>
                    <FormHeading>
                              Request an invite
                    </FormHeading>
                    <form
                              className="form"
                              onSubmit={undefined}
                    >
                              <div>
                                        <Field
                                                  component={[Function]}
                                                  label="Full Name"
                                                  name="name"
                                                  placeholder="Full Name"
                                                  type="text"
                                        />
                                        <Field
                                                  component={[Function]}
                                                  label="Email"
                                                  name="email"
                                                  placeholder="Email"
                                                  type="email"
                                        />
                                        <Field
                                                  component={[Function]}
                                                  label="Confirm email"
                                                  name="confirmEmail"
                                                  placeholder="Confirm email"
                                                  type="email"
                                        />
                              </div>
                              <Button
                                        className="submitBtn"
                                        onClick={undefined}
                                        theme="default"
                                        type="submit"
                              >
                                        Send
                              </Button>
                              <p
                                        className="errorMsg"
                              >
                                        Oops
                              </p>
                    </form>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <InviteForm
    actions={
        Object {
            "requestInvite": [Function],
          }
    }
    error="Oops"
    handleSubmit={[Function]}
    onClose={[Function]}
    submitFailed={true}
    submitSucceeded={false}
    submitted={false}
    submitting={false}
/>,
}
`;

exports[`components/invite-form renders with submitted and submitSucceeded 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <FormSuccess
        onClick={[Function]}
    />
</div>,
  "nodes": Array [
    <div>
      <FormSuccess
            onClick={[Function]}
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <InviteForm
        actions={
                Object {
                        "requestInvite": [Function],
                      }
        }
        error={undefined}
        handleSubmit={[Function]}
        onClose={[Function]}
        submitFailed={false}
        submitSucceeded={true}
        submitted={true}
        submitting={false}
/>,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": InviteForm {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "onSubmit": [Function],
        "props": Object {
          "actions": Object {
            "requestInvite": [Function],
          },
          "error": undefined,
          "handleSubmit": [Function],
          "onClose": [Function],
          "submitFailed": false,
          "submitSucceeded": true,
          "submitted": true,
          "submitting": false,
        },
        "refs": Object {},
        "renderField": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <FormSuccess
                    onClick={[Function]}
          />
</div>,
        "_debugID": 6,
        "_renderedOutput": <div>
          <FormSuccess
                    onClick={[Function]}
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <InviteForm
    actions={
        Object {
            "requestInvite": [Function],
          }
    }
    error={undefined}
    handleSubmit={[Function]}
    onClose={[Function]}
    submitFailed={false}
    submitSucceeded={true}
    submitted={true}
    submitting={false}
/>,
}
`;
